packages/core/src/modules/vc/data-integrity/models/W3cJsonLdVerifiableCredential.ts

```diff
@@ -0,0 +1,42 @@
+import type { LinkedDataProofOptions } from './LinkedDataProof'
+import type { ClaimFormat } from '../../W3cCredentialServiceOptions'
+import type { W3cCredentialOptions } from '../../models/credential/W3cCredential'
+
+import { ValidateNested } from 'class-validator'
+
+import { IsInstanceOrArrayOfInstances, SingleOrArray, asArray } from '../../../../utils'
+import { W3cCredential } from '../../models/credential/W3cCredential'
+
+import { LinkedDataProof, LinkedDataProofTransformer } from './LinkedDataProof'
+
+export interface W3cJsonLdVerifiableCredentialOptions extends W3cCredentialOptions {
+  proof: SingleOrArray<LinkedDataProofOptions>
+}
+
+export class W3cJsonLdVerifiableCredential extends W3cCredential {
+  public constructor(options: W3cJsonLdVerifiableCredentialOptions) {
+    super(options)
+    if (options) {
+      this.proof = Array.isArray(options.proof)
+        ? options.proof.map((proof) => new LinkedDataProof(proof))
+        : new LinkedDataProof(options.proof)
```

Comment:
If we don't we should have a utility method for this.

```typescript
function mapSingleOrArray<T>(v: SingleOrArray<t>, map: (i: T) => SingleOrArray<T>): SingleOrArray<T>
```

and maybe also a `mapSingleOrArrayToArray` which just does the same and always returns an array.

Reply:
Yeah that's a good one, this is messsssy